### ISSUE TITLE
Exporting KML not sending notification to download

### ISSUE THREAD

**Issue by pashiav** [specify/specify7#4854]
**Describe the bug**
When querying a Locality and clicking **Create KML**, the notification _"A notification will appear when the export file is complete and ready for download."_ appears, but no notification is ever received.

Could be related to [https://discourse.specifysoftware.org/t/notifications-tab-not-reactive-anymore/1639/3](https://www.google.com/url?q=https://discourse.specifysoftware.org/t/notifications-tab-not-reactive-anymore/1639/3&sa=D&source=editors&ust=1714408126789921&usg=AOvVaw1yuN3XhUTu5JTAZ0D6PGno)?

**To Reproduce**
1. Go to https://umherb11024-edge.test.specifysystems.org/specify/query/1201/
2. Click on **Query** -> **Create KML**
3. See message  _"A notification will appear when the export file is complete and ready for download."_
4. Wait and see no notification is ever received

**Note**: does not work with selecting a few items for export either

**Expected behavior**
There should be a notification received with a button to download the file.
![Screen Shot 2024-04-29 at 10 56 38 AM](https://github.com/specify/specify7/assets/118556884/ceb2d1f8-c704-48a5-a6c3-77ff57846cad)

**Evidence**

https://github.com/specify/specify7/assets/118556884/31c398d6-0888-49d1-a111-cc3cb8d15b7f

---
OS: macOS Monterey (12.6.5)
Browser: Chrome
Specify 7 Version: `v7.9.3.1` and `edge`
Database Name: `umherb_1_10_24`
[Specify 7 System Information - 2024-04-29T16_00_10.611Z.txt](https://github.com/specify/specify7/files/15151782/Specify.7.System.Information.-.2024-04-29T16_00_10.611Z.txt)

**Comment by CarolineDenis** [comment]
NOTES: 
- cannot reproduce, same db same browser

**Comment by acwhite211** [comment]
spotted this error in the logs when running the kml export
```
specify7-1         | [29/Apr/2024 16:31:39] "POST /stored_query/exportkml/ HTTP/1.0" 200 2
specify7-1         | Exception in thread Thread-101:
specify7-1         | Traceback (most recent call last):
specify7-1         |   File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
specify7-1         |     self.run()
specify7-1         |   File "/usr/lib/python3.8/threading.py", line 870, in run
specify7-1         |     self._target(*self._args, **self._kwargs)
specify7-1         |   File "/opt/specify7/specifyweb/stored_queries/execution.py", line 160, in do_export
specify7-1         |     query_to_kml(session, collection, user, tableid, field_specs, path, spquery['captions'], host,
specify7-1         |   File "/opt/specify7/specifyweb/stored_queries/execution.py", line 249, in query_to_kml
specify7-1         |     coord_cols = getCoordinateColumns(field_specs, table != None)
specify7-1         |   File "/opt/specify7/specifyweb/stored_queries/execution.py", line 267, in getCoordinateColumns
specify7-1         |     f_name = jp[len(jp)-1].name.lower()
specify7-1         | IndexError: tuple index out of range
```

**Comment by acwhite211** [comment]
https://github.com/specify/specify7/blob/8a5f530566d43e7e7c45e65e0e92017961a1f882/specifyweb/stored_queries/execution.py#L261-L288

I made a PR with a quick fix #4855 

We also might want to refactor the export kml process sometime to use the celery worker and make sure to send error notifications when a failure occurs, like in agent merging.
